Python frontend parsing failure

Weird issue in the Python frontend where it appears to be trying to parse a c file using the Python parser, which goes about as well as you would expect. Any idea what might be causing this or where I should look to try and debug this?

```
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing test.py

Type checking warning:
test.py:11: error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]
test.py:12: error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]
Found 2 errors in 1 file (checked 1 source file)

<unknown>:33: SyntaxWarning: invalid escape sequence '\d'
Converting
file /tmp/esbmc.4fd6-b6a5-ff4e/library/builtin_libs.c: Parsing

Type checking warning:
/tmp/esbmc.4fd6-b6a5-ff4e/library/builtin_libs.c:10: error: Unterminated string literal (detected at line 10)  [syntax]
Found 1 error in 1 file (errors prevented further checking)

Traceback (most recent call last):
  File "/tmp/esbmc-python-astgen-7bac-d588-8992/parser.py", line 504, in <module>
    main()
  File "/tmp/esbmc-python-astgen-7bac-d588-8992/parser.py", line 460, in main
    tree = ast.parse(source.read())
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sgtpl/.pyenv/versions/3.12.11/lib/python3.12/ast.py", line 52, in parse
    return compile(source, filename, mode, flags,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<unknown>", line 10
    /* Disable cheri-compressed-cap's debug assertions since they assert that
                                   ^
SyntaxError: unterminated string literal (detected at line 10)
```

Test.py is just the example in the README.md:
```
import random as rand


def div1(cond: int, x: int) -> int:
    if not cond:
        return 42 // x
    else:
        return x // 10


cond: int = rand.random()
x: int = rand.random()

assert div1(cond, x) != 1
```

Hi @sgpthomas,

Which OS are you using?

````
$ esbmc samuel.py
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing samuel.py

Type checking warning:
samuel.py:11: error: Incompatible types in assignment (expression has type "float", variable has type "int")
samuel.py:12: error: Incompatible types in assignment (expression has type "float", variable has type "int")
Found 2 errors in 1 file (checked 1 source file)

Converting
Generating GOTO Program
GOTO program creation time: 1.797s
GOTO program processing time: 0.024s
Starting Bounded Model Checking
Symex completed in: 0.003s (15 assignments)
Caching time: 0.000s (removed 0 assertions)
Slicing time: 0.001s (removed 1 assignments)
Generated 2 VCC(s), 2 remaining after simplification (14 assignments)
No solver specified; defaulting to Boolector
Encoding remaining VCC(s) using bit-vector/floating-point arithmetic
Encoding to solver time: 0.003s
Solving with solver Boolector 3.2.2
Runtime decision procedure: 2.089s
Building error trace

[Counterexample]


State 3 file samuel.py line 6 column 8 function div1 thread 0
----------------------------------------------------
Violated property:
  file samuel.py line 6 column 8 function div1
  division by zero
  x != 0


VERIFICATION FAILED
````

This is my OS:

````
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.5 LTS
Release:        22.04
Codename:       jammy
````

--- a/src/c2goto/cprover_library.cpp
+++ b/src/c2goto/cprover_library.cpp
@@ -282,7 +282,11 @@ void add_cprover_library(contextt &context, const languaget *language)
   if (clib->size == 0)
   {
     if (language)
-      return add_bundled_library_sources(context, *language);
+    {
+      // C library sources must be parsed with C frontend, not the current language
+      std::unique_ptr<languaget> c_lang(new_language(language_idt::C));
+      return add_bundled_library_sources(context, *c_lang);
+    }
     log_error("Zero-lengthed internal C library");
     abort();
   }

